### DISCUSSION TITLE
CLI documentation: Towards beginner friendlyness

### DISCUSSION THREAD

**Discussion by saona-raimundo** [coqui-ai/TTS#1038] [Ideas]
Hi everyone! Thank you for the fantastic work!

I have been playing with Synthesizing Speech TTS (as described [in the docs](https://tts.readthedocs.io/en/latest/inference.html)) and I have some ideas about how to improve the docs, especially for those who "just want to use tts".

Some rough ideas I have are:
- Make it easier to find the docs for "simply synthesize"
  + I found it a bit hard how to just try the tts
- Incorporate usual tts control capabilities to the cli, and mention those before the options to change the model and vocoder.
  + After all my reading around, I am not even sure if one has "usual tts control" for the cli synthesizer.
  + By "usual tts control" I am talking about 
    * Speed (rate)
    * Pitch
    * Volume
  + I know the ongoing work on including [SSML support](#752), but these are much simpler than that
- Explain what a model and a vocoder are, before referring to them.
  + I am sorry if this is too basic, but this section of the docs is supposed to be the one used the most by beginners and I think most (including me) do not know (not yet) what you are refering to with these terms.

**Comment by erogol** [comment]
Thx for posting this. I've a couple of follow-ups. 

> Make it easier to find the docs for "simply synthesize"

Have you checked https://tts.readthedocs.io/en/latest/tutorial_for_nervous_beginners.html ? 

> Incorporate usual tts control capabilities to the cli, and mention those before the options to change the model and vocoder.

`tts -h`  shows you already all the docs like any other command-line tool.  But mentioning these in the docs might be an option 

We don't plan to add these control features any time soon, unless there is a PR that would initiate the effort. 

> Explain what a model and a vocoder are, before referring to them.

Explain model by how? What exactly would you like to know about models, assuming that we cannot go too deep like the papers. 
Have you checked the models' section on https://tts.readthedocs.io/en/latest/? 
Is this enough detail?

**Reply by saona-raimundo** [reply]
Thank you for the answer!

> Have you checked https://tts.readthedocs.io/en/latest/tutorial_for_nervous_beginners.html ?

Yes, I did. I would recommend reverting the order and putting "Synthesizing Speech" before "Training a tts Model", just because this section is for the "nervous beginner" (I know the main focus of coquitts is being able to train tts models). Also, this would "settle the expectations" for what kind of control you can achieve with your trained model.

> tts -h shows you already all the docs like any other command-line tool. But mentioning these in the docs might be an option
We don't plan to add these control features any time soon, unless there is a PR that would initiate the effort.

I am sorry, I thought they were easy to implement after having a trained model and I was just missing the way to access these controls.
I  think that the output of `tts -h` is great. It is a bit unusual to have examples there, but I wish every cli had some :) 

> Explain model by how? What exactly would you like to know about models, assuming that we cannot go too deep like the papers.
Have you checked the models' section on https://tts.readthedocs.io/en/latest/?
Is this enough detail?

Yes, I have. I was puzzled when reading the section "Implemented Models", surely because I lack some background. Maybe I am wrong (please ignore if this is wrong) but I feel that the models listed are not "in the same hierarchy". Let me explain.

There are five sections:
- Text-to-Spectrogram
- End-to-End Models
- Attention Methods
- Speaker Encoder
- Vocoders

After going through the "Tutorial For Nervous Beginners", I understand that you need a "model" together with a "vocoder" to make a complete tts. Looking at the list, it is suggested that any pair of Text-to-Spectrogram and Vocoder can be put together.. but then, some questions arise in my mind:
- What is an End-to-End Models? 
- How do the Attention Methods fit into the picture?

I know these are basic questions, and the solution should not be a big paragraph before the different sections... but maybe a diagram? Text in one side, speech in the other and arrows that go through different steps to get from one to the other?
```dot
digraph {
    Text
    Text-to-Spectrogram
    End-to-End_Models
    Attention_Methods
    Speaker_Encoder
    Vocoders
    Speech
    
    Text -> Text-to-Spectrogram -> Vocoders -> Speech
    Attention_Methods, Speaker_Encoder -> Vocoders
    Text -> End-to-End_Models -> Speech
}
```
(Surely my diagram is wrong)

**Reply by erogol** [reply]
> What is an End-to-End Models?
End-to-End models do not need a vocoder model. They come with a baked vocoder. Like VITS model.

> How do the Attention Methods fit into the picture?
Attention is only a topic for Tacotron models.

**Reply by saona-raimundo** [reply]
> Attention is only a topic for Tacotron models.

So maybe the `Implemented Models` sections should have the following structure?

```
# Component models
## Text-to-Spectrogram
- Tacotron 
- Tacotron2
- Glow-TTS
- ...
### Attention models
Relevant for Tacotron family models only
- Guided Attention
- ...
## Vocoders
- MelGAN
### Speaker Encoder
- GE2E 
# End-to-end models
- VITS
```

This would suggest the dependence of models :)

**Comment by mplattu** [comment]
I've been playing with Coqui-AI few weeks and I can confirm that it is not trivial to get onboard. As the documentation of this project is by far the most user friendly around it attracts many of us without knowledge of the basic concepts.

I would benefit a straightforward guide to develop a model for my  language. So far the most helpful materials have been the `recipes/` and the aforementioned Tutorial for Nervous Beginners. Currently I'm puzzled with following questions:
 * How do you choose between models? Is there a standard process for doing this?
 * So far I've been running the example training scripts (e.g. `TTS/recipes/ljspeech/fast_speech/train_fast_speech.py`). It appears the `best_model.pth.tar` contains what it says, a constantly improving  model. Does the training script end at some point or am I to execute as long as I'm happy with the model?
 * The [docs](https://tts.readthedocs.io/en/latest/training_a_model.html) mention Tensiorboard. Wait, what? I've just been training a model and now I do that again?
 * If the complete training process results both TTS and vocoder models how do you create the latter?

Again, the Coqui-AI is the only project which has taken me this far because of its easy-to-approach CLI and documentation. This might be the very reason you're getting all the attention from us newcomers.

**Reply by erogol** [reply]
> How do you choose between models? Is there a standard process for doing this?

There is no standard procedure, unfortunately. Currently, we assume a basic knowledge of these models. Bit by bit we would make things more friendly for even non-technical users. 

> So far I've been running the example training scripts (e.g. TTS/recipes/ljspeech/fast_speech/train_fast_speech.py). It appears the best_model.pth.tar contains what it says, a constantly improving model. Does the training script end at some point or am I to execute as long as I'm happy with the model?

It goes as long as the number of epochs in the config.

> The docs mention Tensiorboard. Wait, what? I've just been training a model and now I do that again?

You don't need to do repeat something. Just run the tensorboard and that's all. 

> If the complete training process results both TTS and vocoder models how do you create the latter?

Train the vocoder model on the same dataset with the same audio parameters.
